src/pages/home/report/ReportActionItemSingle.js

```diff
@@ -138,9 +138,14 @@ function ReportActionItemSingle(props) {
         if (isWorkspaceActor) {
             showWorkspaceDetails(props.report.reportID);
         } else {
+            // Show participants page IOU report preview
+            if (displayAllActors) {
+                Navigation.navigate(ROUTES.getReportParticipantsRoute(props.iouReport.reportID));
+                return;
+            }
             showUserDetails(props.action.delegateAccountID ? props.action.delegateAccountID : actorAccountID);
         }
-    }, [isWorkspaceActor, props.report.reportID, actorAccountID, props.action.delegateAccountID]);
+    }, [isWorkspaceActor, props.report.reportID, actorAccountID, props.action.delegateAccountID, props.iouReport.reportID, displayAllActors]);
```

Comment:
displayAllActors is not memoized - which mean that on any re-render - current useCallback will be re created all the time.
And also seems deps are missing props.iouReport.reportID' - can you please add it too
Same for isWorkspaceActor and actorAccountID - i know that these are not your changes for isWorkspaceActor and actorAccountID - but can you also fix them